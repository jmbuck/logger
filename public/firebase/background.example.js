// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};