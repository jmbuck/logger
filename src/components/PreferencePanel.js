import React, { Component } from 'react'
import '../css/common.css'

class PreferencePanel extends Component {
    render() {
        return (<div className="Panel">
                <div className="Panel-Container">
                    <div className="Preference-Panel">
                        <div onClick={ () => this.props.history.push("/filters")}>Set Logger Filters</div>
                        <div className="Delete-Account" onClick={ () => this.props.history.push("/deleteaccount")}>Delete Account</div>
                    </div>
                </div>
            </div>
        )
    }
}

export default PreferencePanel