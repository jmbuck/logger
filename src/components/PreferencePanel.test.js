import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PreferencePanel from "./PreferencePanel";
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("PreferencePanel", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should render component", () => {
        const component = Enzyme.shallow(<PreferencePanel />);

        expect(component).toMatchSnapshot();
    });
});

