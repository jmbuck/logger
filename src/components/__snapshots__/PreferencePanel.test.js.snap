// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencePanel should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreferencePanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel-container"
>
        <div
                className="panel-preference"
        >
                <div
                        className="panel-left-nav"
                />
                <div
                        className="panel-center-content"
                >
                        <div
                                onClick={[Function]}
                        >
                                Set Logger Filters
                        </div>
                        <div
                                className="delete-account"
                                onClick={[Function]}
                        >
                                Delete Account
                        </div>
                </div>
                <div
                        className="panel-right-nav"
                >
                        <div
                                className="exit-nav"
                                onClick={[Function]}
                        >
                                <img
                                        src="x.svg"
                                />
                        </div>
                </div>
        </div>
</div>,
      "className": "panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-preference"
>
          <div
                    className="panel-left-nav"
          />
          <div
                    className="panel-center-content"
          >
                    <div
                              onClick={[Function]}
                    >
                              Set Logger Filters
                    </div>
                    <div
                              className="delete-account"
                              onClick={[Function]}
                    >
                              Delete Account
                    </div>
          </div>
          <div
                    className="panel-right-nav"
          >
                    <div
                              className="exit-nav"
                              onClick={[Function]}
                    >
                              <img
                                        src="x.svg"
                              />
                    </div>
          </div>
</div>,
        "className": "panel-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="panel-left-nav"
/>,
            <div
              className="panel-center-content"
>
              <div
                            onClick={[Function]}
              >
                            Set Logger Filters
              </div>
              <div
                            className="delete-account"
                            onClick={[Function]}
              >
                            Delete Account
              </div>
</div>,
            <div
              className="panel-right-nav"
>
              <div
                            className="exit-nav"
                            onClick={[Function]}
              >
                            <img
                                          src="x.svg"
                            />
              </div>
</div>,
          ],
          "className": "panel-preference",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "panel-left-nav",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  onClick={[Function]}
>
                  Set Logger Filters
</div>,
                <div
                  className="delete-account"
                  onClick={[Function]}
>
                  Delete Account
</div>,
              ],
              "className": "panel-center-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Set Logger Filters",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Set Logger Filters",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete Account",
                  "className": "delete-account",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete Account",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="exit-nav"
                onClick={[Function]}
>
                <img
                                src="x.svg"
                />
</div>,
              "className": "panel-right-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  src="x.svg"
/>,
                "className": "exit-nav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "x.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-container"
>
          <div
                    className="panel-preference"
          >
                    <div
                              className="panel-left-nav"
                    />
                    <div
                              className="panel-center-content"
                    >
                              <div
                                        onClick={[Function]}
                              >
                                        Set Logger Filters
                              </div>
                              <div
                                        className="delete-account"
                                        onClick={[Function]}
                              >
                                        Delete Account
                              </div>
                    </div>
                    <div
                              className="panel-right-nav"
                    >
                              <div
                                        className="exit-nav"
                                        onClick={[Function]}
                              >
                                        <img
                                                  src="x.svg"
                                        />
                              </div>
                    </div>
          </div>
</div>,
        "className": "panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="panel-preference"
>
            <div
                        className="panel-left-nav"
            />
            <div
                        className="panel-center-content"
            >
                        <div
                                    onClick={[Function]}
                        >
                                    Set Logger Filters
                        </div>
                        <div
                                    className="delete-account"
                                    onClick={[Function]}
                        >
                                    Delete Account
                        </div>
            </div>
            <div
                        className="panel-right-nav"
            >
                        <div
                                    className="exit-nav"
                                    onClick={[Function]}
                        >
                                    <img
                                                src="x.svg"
                                    />
                        </div>
            </div>
</div>,
          "className": "panel-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="panel-left-nav"
/>,
              <div
                className="panel-center-content"
>
                <div
                                onClick={[Function]}
                >
                                Set Logger Filters
                </div>
                <div
                                className="delete-account"
                                onClick={[Function]}
                >
                                Delete Account
                </div>
</div>,
              <div
                className="panel-right-nav"
>
                <div
                                className="exit-nav"
                                onClick={[Function]}
                >
                                <img
                                                src="x.svg"
                                />
                </div>
</div>,
            ],
            "className": "panel-preference",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "panel-left-nav",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    onClick={[Function]}
>
                    Set Logger Filters
</div>,
                  <div
                    className="delete-account"
                    onClick={[Function]}
>
                    Delete Account
</div>,
                ],
                "className": "panel-center-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Set Logger Filters",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Set Logger Filters",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete Account",
                    "className": "delete-account",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete Account",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="exit-nav"
                  onClick={[Function]}
>
                  <img
                                    src="x.svg"
                  />
</div>,
                "className": "panel-right-nav",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="x.svg"
/>,
                  "className": "exit-nav",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "x.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
